### DISCUSSION TITLE
Serverless function works locally but not showing up when deploying to vercel

### DISCUSSION THREAD

**Discussion by saurabhsharan** [vercel/community#40] [Help] [answered]
I have a file called `checkout.ts` under `pages/api`. When I run `vercel dev` locally the function works fine. However, when I deploy to vercel the function doesn't appear under the "Functions" tab, nor does the function seem to be deployed. Is there anything special I need to do in vercel to enable serverless functions, or is there some other good way to debug this?

**Comment by maumaumaumaumaumau** [comment]
This seems to happen for me too, but with github. I think that may be a vercel's issue, I'm not really sure though.

**Reply by saurabhsharan** [reply]
@maumaumaumaumaumau interesting - did you try contacting vercel support?

**Comment by mcsdevv** [comment]
Hello @saurabhsharan, can you share the public repository so others can take a look?

**Reply by saurabhsharan** [reply]
Sure, it's at https://github.com/saurabhsharan/rove-static

The specific API file is https://github.com/saurabhsharan/rove-static/blob/master/pages/api/checkout.ts

I also have a ticket open with Vercel support - #75319

**Comment by ThisIsBenny** [comment]
i have a similar problem.
I use parameters in my api urls like
GET /api/widgets/12345
POST /api/widgets/12345/approve

and the file structure is
<img width="256" alt="Bildschirmfoto 2020-11-21 um 17 17 32" src="https://user-images.githubusercontent.com/66417262/99881986-7c89a000-2c1d-11eb-99e2-0093962a93d4.png">


This works local with vercel dev and also the last days online. But now, all new deploys are returning the 200.html:
<img width="1678" alt="Bildschirmfoto 2020-11-21 um 17 19 12" src="https://user-images.githubusercontent.com/66417262/99882067-0afe2180-2c1e-11eb-9936-3a480879a04a.png">


I have test a old deployment from 2-3 days ago and there it still works. then i have redeploy this deployment and after the redeployment, i have the same issue :-( 

On localhost, everything is still working.

**Comment by MRVDOG** [comment]
I too am now seeing this issue, last working staging deployment was 6 days ago!
Works on my local env fine.

**Comment by vebits** [comment]
Same issue here. This seems to be a bug with Path Segments as far as I have debugged.

@saurabhsharan Did you get any info from the ticket?

EDIT: Issue still persist when downgrading CLI to vercel@20.1.4

**Reply by vebits** [reply]
@mcsdevv This is quite critical as new changes can't be deployed to production. Please provide any updates here.

**Reply by paulogdm** [reply]
We are investigating the issue as we speak. I will get back to you shortly.

**Reply by paulogdm** [reply]
A fix has been released. Please deploy again and let us know.

**Reply by MRVDOG** [reply]
> A fix has been released. Please deploy again and let us know.

Yup, my functions are now working again, thank you very much :)

**Comment by AdhamHaisami** [comment]
I have the same problem I guess. The app works well locally but none of its functionality works on vercel

**Reply by leerob** [reply]
Could you share a repository with a reproducible example? Thank you!

**Comment by harsh-vish14** [comment]
I am having the same problem and I deployed my next js app from GitHub and cli but both way is not working. Only static pages are working

PLEASE RESOLVE THIS🙏🙏

**Comment by robertwach** [comment]
I have the same issue, I cant make simple HTTP requests, but works perfectly on local env

**Reply by mcsdevv** [reply]
Can you expand on exactly what isn't working here? If you can provide a reproduction of the issue we can take a closer look also.

**Reply by harsh-vish14** [reply]
@mcsdevv in my case getStaticProps and getStaticPaths is not loading page then after the 10-sec application is getting crash
but I deployed on Heroku there it is working fine

**Reply by robertwach** [reply]
I have this file
pages/api/order/index.js

inside I have tried to make requests both

// with axios
axios.post(www.example.com, {}) 


// with node-fetch
fetch(www.example.com, { body: body }

in local environment, it works fine but when deployed at vercel it will not work and when it does it works once after about 5 attempts.

**Reply by robertwach** [reply]
Here is the error from vercel logs 

Client network socket disconnected before secure TLS connection was established

**Reply by mcsdevv** [reply]
Are either of you able to share the relevant code for these files?

**Reply by robertwach** [reply]
@mcsdevv thanks,  I figured the issue though

1. you must use await even if you don't care about the results
2. You cant use fetch(www.example.com).then()

**Comment by Pieparker** [comment, marked as answer]
This discussion was created back in Nov 2020 and a [fix was released for the issue being observed](https://github.com/vercel/vercel/discussions/5442#discussioncomment-136917) back then.

If you are experiencing issues with serverless functions when deployed to Vercel, consider checking out our [docs](https://vercel.com/docs/concepts/functions/serverless-functions) and guides (such as [this one](https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions)) before creating your own discussion here!
